packages/core/test/insured-bridge/e2e/InsuredBridge.e2e.js

```diff
@@ -105,44 +104,50 @@ describe("Insured Bride", () => {
+      assert.equal((await l1Token.balanceOf(l1Wallet.address)).toString(), "0");
+      assert.equal((await l2Token.balanceOf(l2Wallet.address)).toString(), "1234");
     });
     it("Can send tokens back from L2 to L1", async () => {
       // Burn the tokens on L2 and ask the L1 contract to unlock on our behalf.
-      const withdrawTx = await l2StandardBridge.withdraw(l2Erc20.address, 1234, 0, "0x");
+      const withdrawTx = await l2StandardBridge.withdraw(l2Token.address, 1234, 0, "0x");
       await withdrawTx.wait();
 
       // Wait for the message to be relayed to L1. Check addresses in the logs match the L1 token and bridge.
       const [msgHash] = await watcher.getMessageHashesFromL2Tx(withdrawTx.hash);
       const receipt = await watcher.getL1TransactionReceipt(msgHash);
-      assert.equal(receipt.logs[0].address, l1Erc20.address);
+      assert.equal(receipt.logs[0].address, l1Token.address);
       assert.equal(receipt.logs[1].address, l1StandardBridge.address);
 
       // Validate balances on L1 have been incremented accordingly.
-      assert.equal((await l1Erc20.balanceOf(l1Wallet.address)).toString(), "1234");
-      assert.equal((await l2Erc20.balanceOf(l2Wallet.address)).toString(), "0");
+      assert.equal((await l1Token.balanceOf(l1Wallet.address)).toString(), "1234");
+      assert.equal((await l2Token.balanceOf(l2Wallet.address)).toString(), "0");
     });
   });
   describe("Insured bridge functionality", async () => {
-    // Contract objects
-    let l1Finder, l1BridgePoolFactory, l2Timer, l2BridgeDepositBox;
-    beforeEach("Deploy and configure insured bridge contracts on appropriate networks", async () => {
-      let optimisticOracleLiveness = 7200;
-      let proposerBondPct = "0";
-      let identifier = "0x123";
-      let minimumBridgingDelay = 60;
-
-      // Deploy UMA helper contracts
-      l1Finder = await factory__L1_Finder.connect(l1Wallet).deploy();
-      l2Timer = await factory__L2_Timer.connect(l2Wallet).deploy(OPTIMISM_GAS_OPTS);
-      await l2Timer.deployTransaction.wait();
-
-      // TODO: when doing full integration tests we will need the OO as well. For now show that we can do basic bridging.
+    let optimisticOracleLiveness = 7200;
+    let proposerBondPct = "0";
+    let identifier = ethers.utils.formatBytes32String("BRIDGE_TRANSFER_TEST");
+    let minimumBridgingDelay = 60;
+
+    // End to end tested contracts
+    let l1BridgeAdmin, l1BridgePool, l2BridgeDepositBox;
+
+    // UMA ecosystem objects.
+    let l1Timer, l1Finder, l1CollateralWhitelist, l2Timer;
+
+    beforeEach(async () => {
+      // Set up required UMA L1 ecosystem contracts. Note that this also sets up collateralWhitelist, identifierWhitelist
+      // and store which are not used in the tests. TODO: setup OO when we have more implementation cross-chain.
+      ({ l1Timer, l1Finder, l1CollateralWhitelist, l2Timer } = await setUpUmaEcosystemContracts(
```

Comment:
This may not fit exactly but [hardhat fixtures](https://hardhat.org/plugins/hardhat-deploy.html#creating-fixtures) can do similar set up and return values too.

Reply:
ye, that wont work here as we not running hardhat tests :)